### PR TITLE
rh-che #395: Pushing latest and short commit hash images to devshift …

### PR DESCRIPTION
**Pull request by ibuziuk**
### What does this PR do?
Pushing latest and short commit hash images to devshift - https://docs.google.com/document/d/14rKA_Uxve5f_mFNK4vhKhXrMcquiy25AQ5tpHJQwtbc/edit#heading=h.35vszcbdukqa

### What issues does this PR fix or reference?
https://github.com/redhat-developer/rh-che/issues/395

### How have you tested this PR?
have not tested

### CODE REVIEW

File: .ci/cico_deploy.sh [hunk 1]
```diff
@@ -37,19 +37,27 @@ echo "CHE VALIDATION: Verification skipped until job devtools-che-functional-tes
...
+PROD_IMAGE_DEVSHIFT="push.registry.devshift.net/che/che-multiuser:${DEVSHIFT_TAG}"
+PROD_IMAGE_DEVSHIFT_LATEST="push.registry.devshift.net/che/che-multiuser:latest"
 
-if ([ -z "${DEVSHIFT_USERNAME+x}" ] || [ -z "${DEVSHIFT_PASSWORD+x}" ]); then
-    echo "ERROR: Cannot push to registry.devshift.net: credentials are not set. Aborting"
-    exit 1
+if [ -n "${DEVSHIFT_USERNAME}" -a -n "${DEVSHIFT_PASSWORD}" ]; then
+  docker login -u ${DEVSHIFT_USERNAME} -p ${DEVSHIFT_PASSWORD} ${DEVSHIFT_REGISTRY}
+else
+  echo "Could not login to registry.devshift.net, missing credentials for the registry"
```
Comment (benoitf) [inline .ci/cico_deploy.sh]: it should not exit there if credentials are missing ?

Comment (ibuziuk) [reply .ci/cico_deploy.sh]: fixed

File: .ci/cico_deploy.sh [hunk 2]
```diff
@@ -37,19 +37,28 @@ echo "CHE VALIDATION: Verification skipped until job devtools-che-functional-tes
...
 
-echo "CHE VALIDATION: Pushing image ${PROD_IMAGE_DEVSHIFT} to devshift"
+DEVSHIFT_TAG=$(echo $GIT_COMMIT | cut -c1-${DEVSHIFT_TAG_LEN})
+PROD_IMAGE_DEVSHIFT="push.registry.devshift.net/che/che-multiuser:${DEVSHIFT_TAG}"
+PROD_IMAGE_DEVSHIFT_LATEST="push.registry.devshift.net/che/che-multiuser:latest"
 
-if ([ -z "${DEVSHIFT_USERNAME+x}" ] || [ -z "${DEVSHIFT_PASSWORD+x}" ]); then
-    echo "ERROR: Cannot push to registry.devshift.net: credentials are not set. Aborting"
-    exit 1
+if [ -n "${DEVSHIFT_USERNAME}" -a -n "${DEVSHIFT_PASSWORD}" ]; then
```
Comment (l0rd) [inline .ci/cico_deploy.sh]: Why do you need to change the logic here?

File: .ci/cico_deploy.sh [hunk 3]
```diff
@@ -37,19 +37,28 @@ echo "CHE VALIDATION: Verification skipped until job devtools-che-functional-tes
 # . cico_run_EE_tests.sh
 # echo "CHE VALIDATION: Verification passed. Pushing Che server image to prod registry."
 
+echo "CHE VALIDATION: Verification is disabled. Pushing Che server images to prod registry without verification."
+
 STAGE_IMAGE_TO_PROMOTE="${DOCKER_HUB_NAMESPACE}/che-server-multiuser:${CHE_SERVER_DOCKER_IMAGE_TAG}"
-PROD_IMAGE_DEVSHIFT="push.registry.devshift.net/che/che-multiuser:${CHE_SERVER_DOCKER_IMAGE_TAG}"
 
-echo "CHE VALIDATION: Pushing image ${PROD_IMAGE_DEVSHIFT} to devshift"
+DEVSHIFT_TAG=$(echo $GIT_COMMIT | cut -c1-${DEVSHIFT_TAG_LEN})
```
Comment (l0rd) [inline .ci/cico_deploy.sh]: What if `$GIT_COMMIT` is not set? I would rather test that the variable is set as we do for credentials.

File: .ci/cico_deploy.sh (hunk 2 above)
Comment (ibuziuk) [reply .ci/cico_deploy.sh]: because IMO `if / else` structure is more straight forward and easier to understand

File: .ci/cico_deploy.sh (hunk 3 above)
Comment (ibuziuk) [reply .ci/cico_deploy.sh]: ok, will update PR - if `GIT_COMMIT` is not set script will abort

**Comment by ibuziuk** [comment]
@l0rd PR has been updated

**Pull request merged**